ui/pages/add-snap-account/new-snap-account-page/new-snap-account-page.tsx

```diff
@@ -0,0 +1,130 @@
+import { Snap } from '@metamask/snaps-utils';
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { KEY_MANAGEMENT_SNAPS } from '../../../../app/scripts/controllers/permissions/snaps/keyManagementSnaps';
+import { Box, Text } from '../../../components/component-library';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  FlexWrap,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  getSnaps,
+  getsnapsAddSnapAccountModalDismissed,
+} from '../../../selectors';
+import { setSnapsAddSnapAccountModalDismissed } from '../../../store/actions';
+import AddSnapAccountModal from '../add-snap-account-modal';
+import SnapCard from '../snap-card/snap-card';
+
+export interface SnapDetails {
+  id: string;
+  snapId: string;
+  iconUrl: string;
+  snapTitle: string;
+  snapSlug: string;
+  snapDescription: string;
+  tags: string[];
+  developer: string;
+  website: string;
+  auditUrls: string[];
+  version: string;
+  lastUpdated: string;
+}
+
+export interface SnapCardProps extends SnapDetails {
+  isInstalled: boolean;
+  updateAvailable: boolean;
+}
+
+function getShuffledSnapList(arr: SnapDetails[]): SnapDetails[] {
+  return [...arr].map((_, i, arrCopy) => {
+    const rand = i + Math.floor(Math.random() * (arrCopy.length - i));
+    [arrCopy[rand], arrCopy[i]] = [arrCopy[i], arrCopy[rand]];
+    return arrCopy[i];
+  });
+}
+
+export default function NewSnapAccountPage() {
+  const t = useI18nContext();
+  const installedSnaps: Record<string, Snap> = useSelector(getSnaps);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [snapList, setSnapList] = useState(
+    getShuffledSnapList(Object.values(KEY_MANAGEMENT_SNAPS)),
+  );
```

Comment:
Can this be a constant value if the setter is not being used?

Reply:
now set to a constant